base/applications/rapps/appview.cpp

```diff
@@ -1382,7 +1382,21 @@ BOOL CAppsListView::AddInstalledApplication(CInstalledApplicationInfo *InstAppIn
         return FALSE;
     }
 
-    HICON hIcon = (HICON)LoadIconW(hInst, MAKEINTRESOURCEW(IDI_MAIN));
+    /* Load icon from registry */
+    HICON hIcon = NULL;
+    ATL::CStringW szIconPath;
+    if (InstAppInfo->RetrieveIcon(szIconPath))
+    {
+        hIcon = (HICON)ExtractIconW(hInst,
+            szIconPath.GetString(),
+            0);
+    }
+
+    if (!hIcon || GetLastError() != ERROR_SUCCESS)
```

Comment:
GetLastError check should not be here.

Reply:
Fixed, thanks.
Btw I added `GetLastError` check only because it has been done also for available apps:
https://github.com/reactos/reactos/blob/master/base/applications/rapps/appview.cpp#L1418
If that check is not required, shouldn't it also be removed there?